### DISCUSSION TITLE
Announcing LTSP 19.11

### DISCUSSION THREAD

**Discussion by alkisg** [ltsp/ltsp#209] [Q&A]
## Announcing LTSP 19.11

A new version LTSP 19.11 is available in the [PPA](https://ltsp.github.io/docs/ppa).
**Important notice**: please manually install the ltsp-binaries package, now available in the PPA, i.e. run:

```shell
sudo -i
apt update
apt install ltsp-binaries
```

This provides the required ipxe binaries, replacing the previous implementation that fetched them from the Internet via `wget`. This package will only be available in the PPA and not in Debian; by default there, the `ipxe` package will be used instead, which currently misses snponly.efi and memtest.efi.

Changelog from the previous version:

  * Prefer external ltsp-binaries package, fall back to ipxe (#49)
  * Don't exclude /snap (#63)
  * Make RELOGIN work in lightdm (#58)
  * Remove HOSTNAME_PREFIX and _SUFFIX (#57)
  * Enable ltsp.service with compat=10
  * Amend AUTOLOGIN/PASSWORDS_x example (#58)
  * Use SOURCE_DATE_EPOCH to set the date when generating man pages
  * Command line parameters take precedence over ltsp.conf
  * Default to OVERWRITE=1 (#53)
  * Fix ltsp.conf manpage online rendering

**Comment by CharlesMAtkinson** [comment]
After upgrading ltsp to 19.11-1~201911110854~ubuntu18.04.1 and installing ltsp-binaries, I removed /srv/tftp/ltsp/* and ran ltsp ipxe (sorry about the line wraps. "Insert code" did not work as expected. I did not find a hard line end):
`
root:~# ltsp ipxe
Installed /usr/share/ltsp/server/ipxe/ltsp.ipxe in /srv/tftp/ltsp/ltsp.ipxe
Downloading https://github.com/ltsp/binaries/releases/latest/download/memtest.0
LTSP command failed: wget -q https://github.com/ltsp/binaries/releases/latest/download/memtest.0 -O /srv/tftp/ltsp/memtest.0
Aborting ltsp
`
Worked around that by:
`root:~# BINARIES_URL=https://github.com/ltsp/binaries/blob/master/binaries ltsp ipxe `

Also IIUC ltsp ipxe copies files from /usr/share/ltsp/binaries to /srv/tftp/ltsp/.  It would be nice if it preserved the mtime.

**Comment by alkisg** [comment]
I do not see how this is possible.

```shell
cd $(mktemp -d)
wget https://launchpad.net/~ltsp/+archive/ubuntu/ppa/+files/ltsp_19.11-1~201911110854~ubuntu18.04.1_all.deb
dpkg -x ltsp_19.11-1~201911110854~ubuntu18.04.1_all.deb .
grep -r wget || echo NOTFOUND
NOTFOUND
```

I.e. I downloaded the version you're mentioning and there's no `wget` anywhere in the code.

Could you send the output of these commands in your ltsp server?

```shell
grep -rw wget /usr/share/ltsp
apt policy ltsp
```

**Comment by CharlesMAtkinson** [comment]
```shell
root@ltsp4.iciti.av:~# grep -rw wget /usr/share/ltsp
/usr/share/ltsp/server/ipxe/55-ipxe.sh.org:                re wget -q "$BINARIES_URL/$binary" -O "$TFTP_DIR/ltsp/$binary"
root@ltsp4.iciti.av:~# apt policy ltsp
ltsp:
  Installed: 19.11-1~201911110854~ubuntu18.04.1
  Candidate: 19.11-1~201911110854~ubuntu18.04.1
  Version table:
 *** 19.11-1~201911110854~ubuntu18.04.1 500
        500 http://ppa.launchpad.net/ltsp/ppa/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Comment by alkisg** [comment]
At some point in the past, you probably copied 55-ipxe.sh to 55-ipxe.sh.org, and then you forgot to delete it after upgrade, so it essentially overrides the new 55-ipxe.sh with old code.

I.e. just run this: `rm /usr/share/ltsp/server/ipxe/55-ipxe.sh.org`

**Comment by CharlesMAtkinson** [comment]
That was it.  Sorry for the noise.
